web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -465,7 +479,43 @@ class AppConfig extends Component<Props, State> {
 
         const data = await response.json();
         const oldGroups = this.state.configGroups;
-        const newGroups = data.configGroups;
+        const validationErrors: ConfigGroupItemValidationErrors[] =
+          data.validationErrors;
+
+        // track errors at the form level
+        this.setState({ showValidationError: false });
+        let hasValidationError = false;
+
+        // merge validation errors and config group
+        const newGroups = data.configGroups.map((group: ConfigGroup) => {
+          const newGroup = { ...group };
+          const configGroupValidationErrors = validationErrors?.find(
+            (validationError) => validationError.name === group.name
+          );
+
+          if (configGroupValidationErrors) {
+            newGroup.items = newGroup?.items?.map((item: ConfigGroupItem) => {
+              const itemValidationError =
+                configGroupValidationErrors?.item_errors?.find(
+                  (validationError) => validationError.name === item.name
+                );
+
+              if (itemValidationError && item.value) {
+                item.validationError =
+                  itemValidationError?.validation_errors?.[0]?.message;
+                newGroup.hasError = true;
```

Comment:
This is the actual meat of the change. If there is an error then we add it as validationError to the item and hasError to the group.

The rest of the logic is just querying the new validationErrors returned from the API and matching the group.name and item.name that links the validationError item to the group item.